scripts/fbi/hate_crime/download_publication_data.py

```diff
@@ -0,0 +1,363 @@
+    access_url = None
+    if len(year_subitem_list) > 0:
+        for year_subitem in year_subitem_list:
+            if not access_tables_found:
+                all_a = year_subitem.find_all('a')
+                for cur2_a in all_a:
+                    access_url = cur2_a.get('href')
+                    if not access_tables_found:
+                        if cur2_a.string and 'access' in cur2_a.string.lower():
+                            access_tables_found = True
+                            access_url = _get_abs_url(
+                                _STATUS_DICT['cur_year_url'], access_url)
+                        else:
+                            all_b = cur2_a.find_all('b')
+                            for cur_b in all_b:
+                                if not access_tables_found:
+                                    if 'access' in cur_b.string.lower():
+                                        access_tables_found = True
+                                        access_url = _get_abs_url(
+                                            _STATUS_DICT['cur_year_url'],
+                                            access_url)
+    _STATUS_DICT['cur_access_url'] = access_url
+    return access_url
+
+
+def _find_table_html_url() -> list:
+    """Find the URL list of HTML version of all the tables.
+  
+  The method uses status variables set in the status_dict for context.
+  Returns:
+        A list of URLs to the HTML page for all the tables for a particular year.
+  """
+    global _STATUS_DICT
+    # fetch or return from cache
+    if _STATUS_DICT['force_fetch'] or not os.path.isfile(
+            os.path.join(_STATUS_DICT['cache_dir'],
+                         f"{_STATUS_DICT['cur_year']}_access.html")):
+        access_html, access_soup = request_html_soup(
+            _STATUS_DICT['session'], _STATUS_DICT['cur_access_url'])
+        with open(
+                os.path.join(_STATUS_DICT['cache_dir'],
+                             f"{_STATUS_DICT['cur_year']}_access.html"),
+                'w') as fp:
+            fp.write(access_html)
+    else:
+        with open(
+                os.path.join(_STATUS_DICT['cache_dir'],
+                             f"{_STATUS_DICT['cur_year']}_access.html"),
+                'r') as fp:
+            access_html = fp.read()
+            access_soup = BeautifulSoup(access_html, features="html.parser")
+    # convert for use of base URL in the next step
+    _STATUS_DICT['cur_access_url'] = _STATUS_DICT[
+        'cur_access_url'][:_STATUS_DICT['cur_access_url'].rfind('/')]
+
+    tables_div_all = access_soup.find(id='col_b')
+    if not tables_div_all:
+        tables_div_all = access_soup.find(id='col_2')
+    tables_a_all = tables_div_all.find_all('a')
+    if _STATUS_DICT['cur_year'] in _YEARS_WITH_NO_TABLE0:
+        _STATUS_DICT['table_html_url_list'] = ['']
+    else:
+        _STATUS_DICT['table_html_url_list'] = []
+    for i, cur_table_a in enumerate(tables_a_all):
```

Comment:
`i` seems unused in the loop. Can avoid enumerate?

Reply:
fixed. thanks for pointing it out.